### PR TITLE
Always flush responses (#1202)

### PR DESCRIPTION
**Pull request by cesarbs**
#1202

cc @halter73

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 1]
```diff
@@ -782,7 +782,7 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
         }
 
         [Fact]
-        public async Task HeadResponseCanContainContentLengthHeaderButBodyNotWritten()
+        public async Task HeadResponseBodyNotWritten()
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: This test passes in `dev`. Why wouldn't this work?

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs [hunk 2]
```diff
@@ -770,21 +770,21 @@ protected Task ProduceEnd()
         {
             ProduceStart(appCompleted: true);
 
-            // Force flush
-            await SocketOutput.FlushAsync();
```
Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs]: You can remove `ProduceEndAwaited` now. Change 

             if (!HasResponseStarted)
             {
                 return ProduceEndAwaited();
             }

to

             if (!HasResponseStarted)
             {
                 ProduceStart(appCompleted: true);
             }

in ProduceEnd().

**Comment by halter73** [comment]
I actually think I prefer https://github.com/aspnet/KestrelHttpServer/commit/27049598854b8ed14ea3c68ded56404b3ae7f966. It makes HEAD responses flush the header at the same point in the app it would for GET responses. It also avoids creating new async methods.

**Comment by cesarbs** [comment]
@halter73 Ok, I'll close this in favor of your change.

**Pull request closed without merging**